libsoftwarecontainer/unit-test/gateway_unittest.cpp

```diff
@@ -186,16 +202,52 @@ TEST_F(GatewayTest, ConfigEnablesActivateEnablesTeardown) {
 }
 
 /*
- * Test that double activation is not possible
+ * Test that double activation is not possible for gatways that are
+ * not quick-launch enabled.
  */
-TEST_F(GatewayTest, CantActivateTwice) {
+TEST_F(GatewayTest, CanNotActivateTwiceWhenNotQuickLaunchEnabled) {
     loadConfig(validConf);
     ASSERT_TRUE(gw.setConfig(jsonConf));
     ASSERT_TRUE(gw.activate());
     ASSERT_THROW(gw.activate(), GatewayError);
 }
 
 /*
+ * Test that double activation is not possible for gatways that are
+ * quick-launch enabled.
+ */
+TEST_F(GatewayTest, CanNotActivateTwiceWhenIsQuickLaunchEnabled) {
```

Comment:
This test name is not explanatory enough imo. It sounds like we cannot activate this twice because quicklaunch is enabled. But actually we cannot activate it because we did not configure in between.